tests/middleware/test_logging.py

```diff
@@ -7,6 +7,15 @@
 
 from tests.utils import run_server
 from uvicorn import Config
+from uvicorn.protocols.http.h11_impl import H11Protocol
+
+try:
+    from uvicorn.protocols.http.httptools_impl import HttpToolsProtocol
+except ImportError:  # pragma: nocover
+    HttpToolsProtocol = None
+
+
+HTTP_PROTOCOLS = [p for p in [H11Protocol, HttpToolsProtocol] if p is not None]
```

Comment:
```suggestion
HTTP_PROTOCOLS = [p for p in [H11Protocol, HttpToolsProtocol]]
```

Reply:
if we go this way I see no reason for p to be None, see comment below, I'd be in favor for not refgactoring that for just this instance

Reply:
I didn't check, but if the `p is not None` is not there, and `httptools` is not installed, for example, then the list will be `[H11Protocol, None]`, and `None` in the tests will get to `auto`, and then `auto` will choose `h11`. Is that right? 🤔

Reply:
ok your right, so one step back for me,  what is the purpose of those changes ? it's for dev doing stuff on windows and not being able to install httptools (cant remember now if it is possible) or for the CI ? or somnething else

Reply:
It's for users who want to run the test suite against their production environment that doesn't have (or need) `httptools`.

Reply:
> ok your right, so one step back for me, what is the purpose of those changes ? it's for dev doing stuff on windows and not being able to install httptools (cant remember now if it is possible) or for the CI ? or somnething else

The thing is that we were already doing this on the test suite:

https://github.com/encode/uvicorn/blob/b21ecabc5bf911f571e0629438315a1e5472065c/tests/protocols/test_http.py#L20

I guess this was in place so ppl that don't have `httptools` installed could also run the test suite. I think the idea of this PR is to comply with that idea.

Reply:
you're right !